modules/core/04-channel/keeper/upgrade.go

```diff
@@ -109,20 +117,5 @@ func (k Keeper) RestoreChannel(ctx sdk.Context, portID, channelID string, upgrad
 	k.DeleteUpgradeRestoreChannel(ctx, portID, channelID)
 	k.DeleteUpgradeTimeout(ctx, portID, channelID)
 
-	module, _, err := k.LookupModuleByChannel(ctx, portID, channelID)
-	if err != nil {
-		return errorsmod.Wrap(err, "could not retrieve module from port-id")
-	}
-
-	portKeeper, ok := k.portKeeper.(*portkeeper.Keeper)
-	if !ok {
-		panic("todo: handle this situation")
-	}
-
-	cbs, found := portKeeper.Router.GetRoute(module)
-	if !found {
-		return errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module)
-	}
```

Comment:
All this code here to retrieve the application to which we need to callback, would still be needed, right? But one level up? That's it, in the functions where `RestoreChannel` is called, like `ChanUpgradeTimeout`.